### PR TITLE
Em/removedatapair

### PR DESCRIPTION
**Pull request by sgkang**
Remove datapair in Jvec for FDEM and MT
(Should handle memory leackage)

### CODE REVIEW

**Comment by lheagy** [comment]
@sgkang : could you do this pr onto em/dev first since it is em specific? Thanks!

File: SimPEG/EM/TDEM/SrcTDEM.py [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+
+        # # S_e doesn't depend on the model
+
+        # if adjoint:
+        #     return MeSigmaIDeriv( -S_e + C.T * ( MfMui * b ) ).T * v
+
+        # return MeSigmaIDeriv( -S_e + C.T * ( MfMui * b ) ) * v
+
+
+    def S_m(self, prob, time):
```
Comment (rowanc1) [inline SimPEG/EM/TDEM/SrcTDEM.py]: lowercase?

File: SimPEG/EM/TDEM/SrcTDEM.py [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+        S_e = self.S_e(prob, time)
+        return S_m, S_e
+
+    def evalDeriv(self, prob, time, v=None, adjoint=False):
+        if v is not None:
+            return self.S_mDeriv(prob, time, v, adjoint), self.S_eDeriv(prob, time, v, adjoint)
+        else:
+            return lambda v: self.S_mDeriv(prob, time, v, adjoint), lambda v: self.S_eDeriv(prob, time, v, adjoint)
+
+    def S_m(self, prob, time):
```
Comment (rowanc1) [inline SimPEG/EM/TDEM/SrcTDEM.py]: lowercase?

File: SimPEG/EM/TDEM/SrcTDEM.py [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+
+    def __init__(self, offTime=0.):
+        BaseWaveform.__init__(self, offTime, hasInitialFields=True)
+
+    def eval(self, time):
+        raise NotImplementedError('TriangularWaveform has not been implemented, you should write it!')
+
+
+
+class BaseSrc(SimPEG.Survey.BaseSrc):
```
Comment (rowanc1) [inline SimPEG/EM/TDEM/SrcTDEM.py]: We should probably call this `BaseSrc_TDEM` or something more specific.

**Pull request closed without merging**